pkg/rcmanager/ipinstance.go

```diff
@@ -0,0 +1,209 @@
+			ipList = append(ipList, fmt.Sprint(v))
+		}
+		return ipList
+	}()
+	vtepChanged := vtepIP != "" && vtepMac != "" && (vtepIP != remoteVtep.Spec.VtepIP || vtepMac != remoteVtep.Spec.VtepMAC)
+	ipListChanged := remove.Size() == 0 || add.Size() == 0
+	if !newVtep && !ipListChanged && !vtepChanged {
+		return nil
+	}
+
+	if newVtep {
+		remoteVtep, err = m.localClusterRamaClient.NetworkingV1().RemoteVteps().Create(context.TODO(), remoteVtep, metav1.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	} else {
+		remoteVtep, err = m.localClusterRamaClient.NetworkingV1().RemoteVteps().Update(context.TODO(), remoteVtep, metav1.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+	remoteVtep.Status.LastModifyTime = curTime
+	_, err = m.localClusterRamaClient.NetworkingV1().RemoteVteps().UpdateStatus(context.TODO(), remoteVtep, metav1.UpdateOptions{})
+	return err
+}
+
+func (m *Manager) nodeToIPInstance(nodeName string) ([]*networkingv1.IPInstance, error) {
+	podIP, err := m.IPIndexer.ByIndex(ByNodeNameIndexer, nodeName)
+	if err != nil {
+		klog.Warningf("[nodeToPodIP] can't use ipinstance indexer. indexername=%v, nodename=%v, err=%v", ByNodeNameIndexer, nodeName, err)
+		return nil, err
+	}
+	ans := make([]*networkingv1.IPInstance, 0)
+	for _, v := range podIP {
+		if instance, ok := v.(*networkingv1.IPInstance); ok {
+			ans = append(ans, instance)
+		}
+	}
+	return ans, nil
+}
+
+func (m *Manager) nodeToIPList(nodeName string) ([]string, error) {
+	instances, err := m.nodeToIPInstance(nodeName)
+	if err != nil {
+		return nil, err
+	}
+	ipList := make([]string, 0)
+	for _, instance := range instances {
+		if instance.Status.Phase != networkingv1.IPPhaseUsing {
+			continue
+		}
+		ip, _, _ := net.ParseCIDR(instance.Spec.Address.IP)
+		ipList = append(ipList, ip.String())
+	}
+	return ipList, nil
+}
+
+func (m *Manager) RunIPInstanceWorker() {
+	for m.processNextIPInstance() {
+	}
+}
+
+func (m *Manager) processNextIPInstance() bool {
+	obj, shutdown := m.IPQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	err := func(obj interface{}) error {
+		defer m.IPQueue.Done(obj)
+		var (
+			key string
+			ok  bool
+		)
+		if key, ok = obj.(string); !ok {
+			m.IPQueue.Forget(obj)
+			return nil
+		}
+		if err := m.reconcileIPInstance(key); err != nil {
+			// TODO: use retry handler to
+			// Put the item back on the workqueue to handle any transient errors
+			m.IPQueue.AddRateLimited(key)
+			klog.Warningf("[RemoteCluster-IPInstance] failed to reconcileIPInstance. key=%v. err=%v", key, err)
+			return fmt.Errorf("[RemoteCluster-IPInstance] fail to sync '%s' for cluster id=%v: %v, requeuing", key, m.ClusterName, err)
+		}
+		m.IPQueue.Forget(obj)
+		klog.Infof("[RemoteCluster-IPInstance] succeed to sync '%s', cluster=%v", key, m.ClusterName)
+		return nil
+	}(obj)
+
+	if err != nil {
+		klog.Error(err)
+	}
+	return true
+}
+
+func (m *Manager) filterIPInstance(obj interface{}) bool {
+	if !m.GetIsReady() {
```

Comment:
I wrote this deliberately. In my opinion, if remote cluster is not ready, retry is meaningless, we can drop the event directly. After the remote cluster is ready in health checking, it will trigger a reconcile about remote subnet and remote vtep.

Reply:
But the problem is ipinstance reconciling will not be triggered again?